toucan_connectors/databricks/databricks_connector.py

```diff
@@ -0,0 +1,131 @@
+import logging
+from typing import Dict, Optional
+
+import pandas as pd
+import pyodbc
+import requests
+from pydantic import Field, SecretStr, constr
+
+from toucan_connectors.common import ConnectorStatus, pandas_read_sql
+from toucan_connectors.toucan_connector import ToucanConnector, ToucanDataSource
+
+logger = logging.getLogger(__name__)
+
+
+class DatabricksDataSource(ToucanDataSource):
+    query: constr(min_length=1) = Field(
+        ...,
+        description='You can write a query here',
+        widget='sql',
+    )
+
+
+class DataBricksConnectionError(Exception):
+    """ """
+
+
+class DataBricksUnauthorizedError(Exception):
+    """ """
+
+
+class DatabricksConnector(ToucanConnector):
+    data_source_model: DatabricksDataSource
+    host: str = Field(
+        ...,
+        description='The listening address of your databricks cluster',
+        placeholder='my-databricks-cluster.cloudproviderdatabricks.net',
+    )
+    port: int = Field(
+        ..., description='The listening port of your databricks cluster', placeholder=443
+    )
+    http_path: str = Field(
+        ..., description='Databricks compute resources URL', placeholder='sql/protocolv1/o/xxx/yyy'
+    )
+    pwd: SecretStr = Field(
+        None, description='Your personal access token', placeholder='dapixxxxxxxxxxx'
+    )
+    ansi: bool = False
+    on_demand: bool = Field(
+        False, description="If the cluster is auto terminating, wait for it's start before querying"
+    )
+
+    def _build_connection_string(self) -> str:
+        """For constants see: https://docs.databricks.com/integrations/bi/jdbc-odbc-bi.html"""
+        connection_params = {
+            'Driver': '/opt/simba/spark/lib/64/libsparkodbc_sb64.so',  # This path must be correct when installing
+            # ODBC driver
+            'Host': '127.0.0.1' if self.host == 'localhost' else self.host,
+            'Port': self.port,
+            'HTTPPath': self.http_path,
+            'ThriftTransport': 2,
+            'SSL': 1,
+            'AuthMech': 3,
+            'UID': 'token',
+            'PWD': self.pwd.get_secret_value(),
+        }
+        return ';'.join(f'{k}={v}' for k, v in connection_params.items() if v is not None)
+
+    @staticmethod
+    def _get_details(index: int, status: Optional[bool]):
```

Comment:
Could you add typing please ?

Also, maybe the method would be mode explicit in the code if it was called with the check's name rather than its index in a list ?

Reply:
:+1: